log_upload.py

```diff
@@ -0,0 +1,222 @@
+"""
+ASSUMPTIONS:
+    log file is in the SAME directory as this script
+"""
+"""
+CREDIT: Mihir. N for the following implementations
+    -TOML config file reading
+    -global PARSER_URL, SECRET_KEY, OFFLINE_CAN_CHANNEL, OFFLINE_CAN_BITRATE variables
+    -SIGINT_RECVD flag
+    -ANSI sequences
+    -sigint_handler function
+    -parser_request function
+    -process_response function
+"""
+
+import concurrent.futures
+from pathlib import Path
+import sys
+import toml
+from toml import TomlDecodeError
+from typing import Dict
+from datetime import datetime
+import signal
+# import argparse
+import requests
+import glob
+import unicodedata
+
+
+TOML_CONFIG_FILE = Path("./telemetry.toml")
+
+# file existence check
+if not TOML_CONFIG_FILE.is_file():
+    print(f"Unable to find expected TOML config file: \"{TOML_CONFIG_FILE.absolute()}\"")
+    sys.exit(1)
+
+# <----- Read in TOML file ----->
+try:
+    config: Dict = toml.load(TOML_CONFIG_FILE)
+except TomlDecodeError:
+    print(f"Unable to read configuration from {TOML_CONFIG_FILE.absolute()}!")
+    sys.exit(1)
+
+try:
+    PARSER_URL = config["parser"]["url"]
+    SECRET_KEY = config["security"]["secret_key"]
+    OFFLINE_CAN_CHANNEL = config["offline"]["channel"]
+    OFFLINE_CAN_BITRATE = config["offline"]["bitrate"]
+except KeyError:
+    print(f"{TOML_CONFIG_FILE} does not contain expected keys!")
+    sys.exit(1)
+
+# global flag that indicates whether a SIGINT signal was received
+SIGINT_RECVD = False
+
+# ANSI sequences
+ANSI_ESCAPE = "\033[0m"
+ANSI_RED = "\033[1;31m"
+ANSI_GREEN = "\033[1;32m"
+ANSI_YELLOW = "\033[1;33m"
+ANSI_BOLD = "\033[1m"
+
+DEFAULT_MAX_WORKERS = 32
+__PROGRAM__ = "log_upload"
+
+# header to provide with each HTTP request to the parser for API authorization
+AUTH_HEADER = {"Authorization": f"Bearer {SECRET_KEY}"}
+
+DEBUG_WRITE_ENDPOINT = f"{PARSER_URL}/api/v1/parse/write/debug"
+
+# Lengths of messages for differentiating message types
+CAN_LENGTH_MIN      = 20
+CAN_LENGTH_MAX      = 23
+GPS_LENGTH_MIN      = 117
+GPS_LENGTH_MAX      = 128
+IMU_LENGTH_MIN      = 15
+IMU_LENGTH_MAX      = 17
+
+
```

Comment:
There's a lot of code copied / shared between `link_telemetry.py` and this file. To simplify this, I think we should have `link_telemetry.py` loop through each line in the logfile, create the payload (similar to how the "offline upload" script works", and then lets the existing threading and request infrastructure in link_tel handle the rest.

We should think more about how to make the uploading process more user-friendly, and fast. We should be able to only upload data that wasn't written to influx (either because of a parser failure, a influx failure).